gulpfile.js

```diff
@@ -386,6 +386,38 @@ function thirdPartyBootstrap(watch, shouldMinify) {
 var activeBundleOperationCount = 0;
 
 /**
+ * Synchronously concatenates the given files into the given destination
+ *
+ * @param {string} destFilePath File path to write the concatenated files to
+ * @param {Array<string>} files List of file paths to concatenate
+ */
+function concatFiles(destFilePath, files) {
+	var all = files.map(function(filePath) {
+		return fs.readFileSync(filePath, 'utf-8');
+	});
+
+	fs.writeFileSync(destFilePath, all.join('\n'), 'utf-8');
+}
+
+/**
+ * Allows (ap|pre)pending to the already compiled, minified JS file
+ *
+ * @param {string} srcFilename Name of the JS source file
+ * @param {string} destFilePath File path to the compiled JS file
+ */
+function appendToCompiledFile(srcFilename, destFilePath) {
+  if (srcFilename == 'amp-viz-vega.js') {
+    // Prepend minified d3 and vega third_party to compiled amp-viz-vega.js
+    concatFiles(destFilePath, [
```

Comment:
Concating JS files can be dangerous. We need to either test that the previous file ends with `;`, or the next files starts with it.